src/django_components/utils.py

```diff
@@ -1,41 +1,4 @@
-import glob
-from pathlib import Path
-from typing import Any, Callable, List, NamedTuple, Optional
-
-from django.template.engine import Engine
-
-from django_components.template_loader import Loader
-
-
-class SearchResult(NamedTuple):
-    searched_dirs: List[Path]
-    matched_files: List[Path]
-
-
-def search(search_glob: Optional[str] = None, engine: Optional[Engine] = None) -> SearchResult:
```

Comment:
I've split this function to `get_dirs` and `search_dirs` and moved them to `autodiscover.py`